src/dotnet/Common/Services/Events/AzureEventGridEventService.cs

```
@@ -0,0 +1,361 @@
+        private EventGridClient? _eventGridClient;
+
+        private readonly TimeSpan _eventProcessingHeartbeat = TimeSpan.FromMinutes(1);
+
+        private readonly Dictionary<string, EventSetEventDelegate?> _eventSetEventDelegates = new()
+        {
+            {
+                EventSetEventNamespaces.FoundationaLLM_ResourceProvider_Agent,
+                null
+            }
+        };
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="AzureEventGridEventService"/> event service.
+        /// </summary>
+        /// <param name="settingsOptions">The options providing the settings for the service.</param>
+        /// <param name="profileOptions">The options providing the profile for the service.</param>
+        /// <param name="logger">The logger used for logging.</param>
+        public AzureEventGridEventService(
+            IOptions<AzureEventGridEventServiceSettings> settingsOptions,
+            IOptions<AzureEventGridEventServiceProfile> profileOptions,
+            ILogger<AzureEventGridEventService> logger)
+        {
+            _settings = settingsOptions.Value;
+            _profile = profileOptions.Value;
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                _eventGridClient = GetClient();
+                if (_eventGridClient == null)
+                    _logger.LogCritical("The Azure Event Grid client could not be initialized. The Azure Event Grid event service will not listen for any events.");
+                else
+                    _logger.LogInformation("The Azure Event Grid event service was successfully initialized.");
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "The Azure Event Grid client could not be initialized. The Azure Event Grid event service will not listen for any events.");
```

Are we logging and swallowing this exception because the Event Grid service is optional? I suppose this makes sense for existing deployments, but do we want to raise the exception in general to ensure we always have this service in a functional state?

---

Reviewed the startup sequence and raised an exception is start fails, delegating the decision to the hosting service. In the case of `CoreAPI`, the Event Service is considered optional, so the exception will not prevent it from starting.